### PR TITLE
XSTUSD-based pairs

### PR DESCRIPTION
**Pull request by wer1st**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -4,5 +4,6 @@ def pipeline = new org.docker.AppPipeline(steps: this,
     dockerImageName: 'sora2/pricing-server',
     dockerRegistryCred: 'bot-sora2-rw',
     secretScannerExclusion: '.*docker-compose.yml',
+    dockerImageTags: ['xstusd-pairs':'xstusd'],
```
Comment (wer1st) [inline Jenkinsfile:7]: Will be removed

File: processing.py [hunk 2]
```diff
@@ -75,20 +77,54 @@ def process_swap_transaction(timestamp, extrinsicEvents, ex_dict):
...
+                    get_value(param)["WithDesiredInput"], "min_amount_out", 1)
+            else:  # then we do it by desired output
+                output_amount = get_by_key_or_index(
+                    get_value(param)["WithDesiredOutput"], "desired_amount_out", 0)
+                input_amount = get_by_key_or_index(
+                    get_value(param)["WithDesiredOutput"], "max_amount_in", 1)
+        elif param["name"] == "selected_source_types":
+            filter_mode = get_value(param) or ["SMART"]
+
+    logging.info(">>> my_logs: process_swap_transaction dex = %i", dex_id)
```
Comment (wer1st) [inline processing.py:103]: Will be removed

File: run_node_processing.py [hunk 3]
```diff
@@ -280,31 +283,62 @@ def get_end(substrate: SubstrateInterface):
...
+                                (
+                                    xor_id_int,
+                                    intermediate_amount,
+                                    to_asset,
+                                    tx.pop("out_amount"),
+                                ),
+                            ]
+                    if dex_id == 1:
+                        if from_asset == xstusd_id_int or to_asset == xstusd_id_int:
+                            logging.info(">>> my_logs: block = %i, dex = 1, from/to asset suitable, from = 0x%x, to = 0x%x, in = %i, out = %i", block, from_asset, to_asset, tx["in_amount"], tx["out_amount"])
```
Comment (wer1st) [inline run_node_processing.py:317]: Will be removed

File: run_node_processing.py [hunk 4]
```diff
@@ -280,31 +283,62 @@ def get_end(substrate: SubstrateInterface):
...
+                            data = [
+                                (
+                                    from_asset,
+                                    tx.pop("in_amount"),
+                                    to_asset,
+                                    tx.pop("out_amount"),
+                                )
+                            ]
+                        else:
+                            logging.info(">>> my_logs: block = %i, dex = 1, from/to asset not suitable, from = 0x%x, to = 0x%x, in = %i, out = %i, intermediate_amount = %i", block, from_asset, to_asset, tx["in_amount"], tx["out_amount"], intermediate_amount)
```
Comment (wer1st) [inline run_node_processing.py:327]: Will be removed

File: run_node_processing.py [hunk 5]
```diff
@@ -448,20 +483,33 @@ def get_end(substrate: SubstrateInterface):
                                             amount=val_reminted_parliament,
                                         )
                                     )
+            new_swaps = []
```
Comment (wer1st) [inline run_node_processing.py:486]: Will be removed

File: run_node_processing.py [hunk 6]
```diff
@@ -448,20 +483,33 @@ def get_end(substrate: SubstrateInterface):
...
+                    new_swaps.append(swap[3])
+                if swap[0] == 1:
+                    other_asset = swap[1] if swap[2] == xstusd_id_int else swap[2]
+                    other_asset = '{0:#0{1}x}'.format(other_asset, 66)
+                    params = [1, XSTUSD_ID, other_asset, '1000000000000000000', 'WithDesiredInput', [], 'Disabled', block_hash]
+                    result = substrate.rpc_request('liquidityProxy_quote', params)
+                    pair = pairs[swap[1], swap[2]]
+                    amount = int(result['result']['amount']) / DENOM
+                    pair.quote_price = amount if swap[1] == xstusd_id_int else 1 / amount
+                    logging.info(">>> my_logs: swap enum dex = 1, pair = %s, value = %s", pair, swap)
```
Comment (wer1st) [inline run_node_processing.py:506]: Will be removed

**Review by N1ghtStorm — APPROVED** [review]

**Review by vovac12 — APPROVED** [review]

**Pull request merged**